doc/source/notebooks/advanced/varying_noise.pct.py

```diff
@@ -114,10 +115,16 @@ def variational_expectations(self, Fmu, Fvar, Y):
 def objective_closure():
     return - model.log_marginal_likelihood()
 
-opt = gpflow.optimizers.Scipy()
-opt.minimize(objective_closure,
-             model.trainable_variables,
-             options=dict(maxiter=ci_niter(1000)))
+natgrad = NaturalGradient(gamma=1.0)
+adam = tf.optimizers.Adam()
+
+from gpflow.utilities import set_trainable
+set_trainable(model.q_mu, False)
+set_trainable(model.q_sqrt, False)
```

Comment:
Could you add a comment explaining why we do this (so that the Adam optimizer doesn't keep moving the variational parameters as well)? (and would the import be better at the top of the file perhaps?)
I think that is important to people new to natural gradients and the way the optimizers work.

Reply:
Would
> We mark the variational parameters as not trainable so that they are not included in the `model.trainable_variables` when we optimize using the Adam optimizer. We train the variational parameters separately using the natural gradient method.

Reply:
Yes, that's great!